src/DynamoCore/Graph/Annotations/AnnotationModel.cs

```diff
@@ -243,33 +291,94 @@ public NodeModel PinnedNode
             }
         }
 
+        private double widthAdjustment;
+        /// <summary>
+        /// Adjustment set to the size of the groups
+        /// width.
+        /// </summary>
+        public double WidthAdjustment
+        {
+            get { return widthAdjustment; }
+            set 
+            {
+                if (value == widthAdjustment) return;
+                widthAdjustment = value;
+                UpdateBoundaryFromSelection();
+            }
+        }
+
+        private double userHeightAdjustment;
+        /// <summary>
+        /// Adjustment set to the size of the groups
+        /// height.
+        /// </summary>
+        public double HeightAdjustment
+        {
+            get { return userHeightAdjustment; }
+            set 
+            {
+                if (value == userHeightAdjustment) return;
+                userHeightAdjustment = value;
+                UpdateBoundaryFromSelection();
+            }
+        }
+
+        private bool isExpanded = true;
+        /// <summary>
+        /// Returns whether or not the group is expanded.
+        /// </summary>
+        public bool IsExpanded 
+        {
+            get { return isExpanded; }
+            set
+            {
+                isExpanded = value;
+                UpdateBoundaryFromSelection();
+            }
+        }
+
+        /// <summary>
+        /// Checks if this group contains any nested groups.
+        /// </summary>
+        public bool HasNestedGroups => nodes.OfType<AnnotationModel>().Any();
+
         #endregion
 
         /// <summary>
         /// Initializes a new instance of the <see cref="AnnotationModel"/> class.
         /// </summary>
         /// <param name="nodes">The nodes.</param>
         /// <param name="notes">The notes.</param>
-        public AnnotationModel(IEnumerable<NodeModel> nodes, IEnumerable<NoteModel> notes)
-        {                                 
-            var nodeModels = nodes as NodeModel[] ?? nodes.ToArray();           
+        public AnnotationModel(IEnumerable<NodeModel> nodes, IEnumerable<NoteModel> notes) : this(nodes, notes, new List<AnnotationModel>()) { }
+
+        public AnnotationModel(IEnumerable<NodeModel> nodes, IEnumerable<NoteModel> notes, IEnumerable<AnnotationModel> groups)
```

Comment:
Would you add comments for this new constructor? Usually for constructors, it's must-have